src/com/trilead/ssh2/packets/PacketKexDHInit.java

```diff
@@ -12,11 +10,11 @@
 {
 	byte[] payload;
 
-	BigInteger e;
+	byte[] publicKey;
 
-	public PacketKexDHInit(BigInteger e)
+	public PacketKexDHInit(byte[] publicKey)
```

Comment:
Another API breaker. Consider keeping the BigInteger and converting to/from byte, or clone the class and migrate internal usage.